src/main/java/com/paneedah/weaponlib/core/WeaponlibClassTransformer.java

```diff
@@ -757,8 +774,8 @@ else if (layerHeldItemClassInfo != null
                     && entityLivingBaseClassInfo.methodMatches("attackEntityFrom", "(Lnet/minecraft/util/DamageSource;F)Z", classname, name, desc)) {
                 return new AttackEntityFromMethodVisitor(cv.visitMethod(access, name, desc, signature, exceptions));
             } else if (entityLivingBaseClassInfo != null
-                    && entityLivingBaseClassInfo.methodMatches("attackEntityFrom", "(Lnet/minecraft/util/DamageSource;F)Z", classname, name, desc)) {
-                return new AttackEntityFromMethodVisitor(cv.visitMethod(access, name, desc, signature, exceptions));
+                    && entityLivingBaseClassInfo.methodMatches("knockBack", "(Lnet/minecraft/entity/Entity;FDD)V", classname, name, desc)) {
+                return new KnockBackMethodVisitor(cv.visitMethod(access, name, desc, signature, exceptions));
```

Comment:
Why?

Reply:
I see that the knockback algorithm will halve the motion of the original entity